### PR TITLE
TVpaint communication enhancement and creators

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changes
- `stdout`, `stderr` and `stdin` are not set when launching TVPaint as subprocess, because process got stuck after specific amount of websocket callbacks (maybe has something to do with usage of stdout in plugin)
- replaced `MainThreadChecker` QThread with simple QTimer
- Skipped returning of result from tools callback as they may return QWidget which is not possible to send as response to websocket request.
- Added `Creator` and `Loader`.
- Added few pipeline and lib functions for executing george scripts and getting metadata from workfile.
    - Should handle storing instances to publish.
        - Added contained key`"uuid"` where `uuid4` value  is stored to help identify which instance is which.
    - Metadata are stored in json format but quotes are replaced with placeholders because george script can removed then during string operations (`'` -> `{__sq__}` and `"` -> `{__dq__}`)
    - A lot of them require to execute complex george scripts with an output. Only solution I found to be able do that was to store output to a temporary file inside the george script with command `tv_writetextfile ...`.
    - `tv_writetextfile` command will show popup when executed first time. Because of this is this command executed as first thing on TVPaint launch which is not user friendly. It is possible to show the popup "when needed" but most of the time is popup under our tools so it seems it's stuck as nothing happens.

|:black_flag: |this depends on|
|---|---|
|pype|https://github.com/pypeclub/pype/pull/693|

### CODE REVIEW

File: avalon/tvpaint/pipeline.py [hunk 1]
```diff
@@ -28,8 +36,145 @@ def uninstall():
 
     """
 
-    pyblish.deregister_host("tvpaint")
+    pyblish.api.deregister_host("tvpaint")
+
+
+def containerise(name, namespace, nodes, context, loader):
```
Comment (mkolar) [inline avalon/tvpaint/pipeline.py:42]: You are claiming schema `schema": "avalon-core:container-2.0` here but not adhering to it. Id is there so we can differentiate between what is container and what is publishable instance. ObjectName in TV paint case doesn't make sense and should be replaced by UUID as you mentioned
```
  "required": [
        "schema",
        "id",
        "objectName",
        "name",
        "namespace",
        "loader",
        "representation"
    ],
```

File: avalon/tvpaint/pipeline.py [hunk 2]
```diff
@@ -28,8 +36,145 @@ def uninstall():
...
 
 def ls():
-    return []
+    return workfile_metadata(SECTION_NAME_CONTAINERS)
+
+
+class Creator(api.Creator):
+    def __init__(self, *args, **kwargs):
+        super(Creator, self).__init__(*args, **kwargs)
+        # Replace unimportant `id` value "pyblish.avalon.instance" with
```
Comment (mkolar) [inline avalon/tvpaint/pipeline.py:130]: I'd prefer if we keep this consistent with other hosts, of course we need uuid, but add "id" as we do for other hosts as well.

**Review by mkolar — CHANGES_REQUESTED** [review]

File: avalon/tvpaint/pipeline.py (hunk 1 above)
Comment (iLLiCiTiT) [reply avalon/tvpaint/pipeline.py]: I just copied containerise from another host, removed few lines and then didn't get back to it as was focusing on creators. Added `"id"` and `"objectName"` is layer id (which is only reference to loaded stuff).

File: avalon/tvpaint/pipeline.py (hunk 2 above)
Comment (iLLiCiTiT) [reply avalon/tvpaint/pipeline.py]: It's ok for me just didn't see reason why to add more data to each item `"pyblish.avalon.container"` when is never used.

File: avalon/tvpaint/pipeline.py (hunk 1 above)
Comment (iLLiCiTiT) [reply avalon/tvpaint/pipeline.py]: I'm not sure if layer id is enough but I think 2 loaded images can't have same layer so it is actually good identifier. Will be possible that one loaded "item" will have more layers (audio 2 channels)? in that case it should be `"layer_ids"` with multiple layers.

**Pull request merged**